File: app/src/main/java/com/example/myapplication4/Clases/Ingreso.java
package com.example.myapplication4.Clases;

import java.math.BigDecimal;
import java.time.LocalDate;

public class Ingreso extends Transaccion {

    public Ingreso(int id, String desc, BigDecimal monto, LocalDate fecha, Categoria categoria){
        super(id, desc, monto, fecha, categoria);
    }
}

File: app/src/main/java/com/example/myapplication4/Clases/Ingreso.kt
package com.example.myapplication4.Clases
import java.time.LocalDate

// Clase hija: Ingreso
data class Ingreso(
    val idIngreso: Int,
    val descIngreso: String,
    val montoIngreso: Float,
    val fechaIngreso: LocalDate,
    val categoriaIngreso: Categoria
) : Transaccion(idIngreso, descIngreso, montoIngreso, fechaIngreso, categoriaIngreso)